NeuroIDAdvancedDevice/AdvancedDevice.swift

```diff
@@ -12,22 +12,26 @@ struct NIDResponse: Codable {
     let key: String
 }
 
-public class NeuroIDADV: NSObject {
-    public static func getAdvancedDeviceSignal(
-        _ apiKey: String,
-        completion: @escaping (Result<String, Error>) -> Void
-    ) {
-        getAPIKey(apiKey) { result in
-            switch result {
-            case .success(let fAPiKey):
-                retryAPICall(apiKey: fAPiKey, maxRetries: 3, delay: 2) { result in
-                    completion(result)
-                }
-            case .failure(let error):
-                completion(.failure(error))
-            }
-        }
-    }
+/** Interface that allows for testing */
+protocol DeviceSignalService {
+    func getAdvancedDeviceSignal(_ apiKey: String, completion: @escaping (Result<String, Error>) -> Void)
+}
+
+public class NeuroIDADV: NSObject, DeviceSignalService {
+    public static let shared = NeuroIDADV()
```

Comment:
Is this ever used? Also why would a class have a attribute that is a value of itself?

Reply:
This is singleton that will be the way we access the NeuroIDADV class `NeuroIDADV.shared` It will be used in other SDK. Singleton needed in order to implement protocols.